### PR TITLE
Fix for Issue 193

### PR DESCRIPTION
**Pull request by PsypherPunk**
Fix for the `NullPointerException` when using a `RemoteResourceIndex`.

Uses a `ClosestResultTrackingFilter`, much the same as the `LocalResourceIndex`, which doesn't appear to have this problem.

### CODE REVIEW

**Comment by anjackson** [comment]
Looks good to me. Okay for you @kris-sigur?

**Comment by kris-sigur** [comment]
Yes. Looks good.

File: wayback-core/src/main/java/org/archive/wayback/resourceindex/RemoteResourceIndex.java [hunk 1]
```diff
@@ -124,8 +127,11 @@ public SearchResults query(WaybackRequest wbRequest)
 		ResourceNotInArchiveException, BadQueryException,
 		AccessControlException {
 //		throw new ResourceIndexNotAvailableException("oops");
-		return urlToSearchResults(getRequestUrl(wbRequest),
+        closestGroup = new ClosestTrackingCaptureFilterGroup(wbRequest, canonicalizer);
```
Comment (ato) [inline wayback-core/src/main/java/org/archive/wayback/resourceindex/RemoteResourceIndex.java]: Unless I'm missing something, this is a race condition as closestGroup is effectively a global variable here. A second thread can jump in and change closestGroup to a different request before its accessed below.

**Comment by PsypherPunk** [comment]
@kngenie commented on the Wayback [issue](https://webarchive.jira.com/browse/ACC-117): it's a configuration/documentation problem.

The `closestSelector` needs initialising (in the `ArchivalUrlReplay.xml`, according to Colin):

    <property name="closestSelector">
        <bean class="org.archive.wayback.archivalurl.ArchivalUrlDateRedirectingClosestResultSelector"/>
    </property>

**Pull request merged**